aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -535,27 +578,25 @@ void flip_kernel(TensorIterator& iter, const bool quantized) {
         });
     });
   } else {
-    // Special case: horizontal flip with vectorization and input is contiguous
-    // Context: horizontal flip leads to strides[0] < 0 and
-    // thus is_contiguous condition is not satisfied and non-vectorized code path is taken.
     auto output_strides = iter.strides(0);
     auto input_strides = iter.strides(1);
-    if (iter.ndim() > 0 && output_strides[0] < 0 && input_strides[0] == iter.element_size(1)) {
+    if (iter.ndim() > 0 && output_strides[0] == -iter.element_size(0) && input_strides[0] == iter.element_size(1)) {
+      // Special case: horizontal flip with vectorization and input is contiguous
+      // Context: horizontal flip leads to strides[0] < 0 and
+      // thus is_contiguous condition is not satisfied and non-vectorized code path is taken.
       auto iter_dtype = iter.dtype();
-      if (iter_dtype == kByte) {
-        return cpu_hflip_vec<uint8_t>(iter);
-      } else if (iter_dtype == kFloat) {
-        return cpu_hflip_vec<float>(iter);
-      } else if (iter_dtype == kInt) {
-        return cpu_hflip_vec<int32_t>(iter);
-      } else if (iter_dtype == kShort) {
-        return cpu_hflip_vec<int16_t>(iter);
-      } else if (iter_dtype == kLong) {
-        return cpu_hflip_vec<int64_t>(iter);
-      } else if (iter_dtype == kDouble) {
-        return cpu_hflip_vec<double>(iter);
-      }
-      // other dtypes are handled below with cpu_kernel_vec
+      // Ignoring half and bfloat16 as cpu_hflip_vec is slower than cpu_kernel_vec
+      if (isIntegralType(iter_dtype, true) || iter_dtype == kDouble || iter_dtype == kFloat) {
+        AT_DISPATCH_ALL_TYPES_AND(kBool,
+            iter_dtype, "hflip_cpu", [&iter] {
+              cpu_hflip_vec<scalar_t>(iter);
+        });
+        return;
+      }
+      // other dtypes (float16, bfloat16, complex) are handled by cpu_kernel_vec (see below)
+    } else if (iter.has_contiguous_first_dim()) {
+      // Special case: vertical flip using memcpy (faster than generic cpu_kernel_vec)
+      return cpu_vflip_memcpy(iter);
```

Comment:
Ideally this change should be supported with a range of benchmarks on both AVX2 and AVX512. Particularly if memcpy doesn't use AVX512.

Reply:
@peterbell10 I updated PR's description with benchmarks for hflip and vflip on avx2 and avx512.